services/catalog/src/simcore_service_catalog/core/background_tasks.py

```diff
@@ -197,20 +165,25 @@ async def _ensure_published_templates_accessible(
         await services_repo.upsert_service_access_rights(missing_services_access_rights)
 
 
-async def _sync_services_task(app: FastAPI) -> None:
+async def _run_sync_services(app: FastAPI):
+    # FIXME: need to guarantee that these are in place!
     default_product: Final[str] = app.state.default_product_name
     engine: AsyncEngine = app.state.engine
 
+    # check that the list of services is in sync with the registry
+    await _ensure_registry_and_database_are_synced(app)
+
+    # check that the published services are available to everyone
+    # (templates are published to GUESTs, so their services must be also accessible)
+    await _ensure_published_templates_accessible(engine, default_product)
+
+
+async def _sync_services_task(app: FastAPI) -> None:
     while app.state.registry_syncer_running:
         try:
             _logger.debug("Syncing services between registry and database...")
 
-            # check that the list of services is in sync with the registry
-            await _ensure_registry_and_database_are_synced(app)
-
-            # check that the published services are available to everyone
-            # (templates are published to GUESTs, so their services must be also accessible)
-            await _ensure_published_templates_accessible(engine, default_product)
+            await _run_sync_services(app)
 
```

Comment:
we have now the https://github.com/ITISFoundation/osparc-simcore/blob/194c626059600ff1a2cec44ff43d45a5eda93c86/packages/service-library/src/servicelib/background_task.py, that could be used instead of this maybe?

Reply:
@sanderegg ABSOLUTELY! I totally agree. This was my idea for the coming PRs but I do not want to do all at once. For that reason first I added a test to this ... which BTW had none :-(